dotnet/src/SemanticKernel.Core/Data/TextSearchStore/TextSearchDocument.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.SemanticKernel.Data;
+
+/// <summary>
+/// Represents a document that can be used for Retrieval Augmented Generation (RAG).
+/// </summary>
+[Experimental("SKEXP0130")]
+public sealed class TextSearchDocument
+{
+    /// <summary>
+    /// Gets or sets an optional list of namespaces that the document should belong to.
+    /// </summary>
+    /// <remarks>
+    /// A namespace is a logical grouping of documents, e.g. may include a group id to scope the document to a specific group of users.
+    /// </remarks>
+    public List<string> Namespaces { get; set; } = [];
```

Comment:
Public members ideally suggest using interfaces.

```suggestion
    public IList<string> Namespaces { get; set; } = [];
```

Reply:
What is the benefit here?